registry-library/library/library.go

```diff
@@ -0,0 +1,190 @@
+package library
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path"
+	"text/tabwriter"
+	"time"
+
+	"github.com/containerd/containerd/remotes/docker"
+	"github.com/deislabs/oras/pkg/content"
+	"github.com/deislabs/oras/pkg/oras"
+	indexSchema "github.com/devfile/registry-support/index/generator/schema"
+)
+
+const (
+	// Devfile media types
+	DevfileConfigMediaType  = "application/vnd.devfileio.devfile.config.v2+json"
+	DevfileMediaType        = "application/vnd.devfileio.devfile.layer.v1"
+	DevfileVSXMediaType     = "application/vnd.devfileio.vsx.layer.v1.tar"
+	DevfileSVGLogoMediaType = "image/svg+xml"
+	DevfilePNGLogoMediaType = "image/png"
+	DevfileArchiveMediaType = "application/x-tar"
+
+	scheme                = "http"
+	httpRequestTimeout    = 30 * time.Second // httpRequestTimeout configures timeout of all HTTP requests
+	responseHeaderTimeout = 30 * time.Second // responseHeaderTimeout is the timeout to retrieve the server's response headers
+)
+
+var (
+	stacksPath = os.Getenv("DEVFILE_STACKS")
+	indexPath  = os.Getenv("DEVFILE_INDEX")
+)
+
+// GetRegistryIndex gets the index of the specified registry
+func GetRegistryIndex(registry string) ([]indexSchema.Schema, error) {
```

Comment:
I think we should call this either `GetRegistryStacks` (or to `ListRegistryStacks` and change the function below to `PrintRegistryStacks`) instead as it's more clear what it does than `GetRegistryIndex`.

Reply:
I am not sure if `GetRegistryStacks/ListRegistryStacks` is more clear to explain the functionality of this function as this function is only return the content of the registry index, maybe `GetRegistryIndex` is already self-explanatory.

Reply:
It's a bit of a nit and yes, that name is self explanatory, but I'm thinking of the library adopters though or anyone trying to get the list of stacks from a registry. To get the list of stacks, this function (GetRegistryIndex) index has to be called. If someone's not entirely familiar with what an index.json or index entry is, they may not know that this function returns the list of stacks.

Reply:
Like I said though, it's a nit, so if you prefer `GetRegistryIndex` I'm fine with that.

Reply:
I see, if that's the case, I can rename the function to `GetRegistryStacks/ListRegistryStacks`, and explain the functionality (eg. related to index.json/index) more in the comment so that it's more friendly for consumer who is not familiar with registry structure and includes the functionality details as well.